### DISCUSSION TITLE
Can't debug findOne results in VSCode when instance is populated with related OneToMany / ManyToMany relation.

### DISCUSSION THREAD

**Discussion by sebastiaanviaene** [mikro-orm/mikro-orm#5525] [General]
### Describe the bug

- Since we switched to Mikro v6 we noticed some very weird behaviour when running `findOne` queries.
- This only happens in VSCode as far as we know. 
- When I try to hover over the result it does not show anything 
- When I try to open the local variable in the sidebar of VSCode it keeps loading
- The node process responsible for the debugger also spikes to 100% CPU and is clearly stuck
- Not populating the instance works and using `find` to get a list also works
- My colleague tried using WebStorm / Chrome debugger and there it works without issues

![image](https://github.com/mikro-orm/mikro-orm/assets/6652818/767c4ca3-2eeb-4cb7-8157-fb198a125c6c)
![image](https://github.com/mikro-orm/mikro-orm/assets/6652818/78752c24-250a-400d-ae36-2f277770f7ca)

I've created a repo forking the reproduction one and with minimal changes was able to reproduce the behaviour (as you can see in the image above)

### Reproduction

Forked from `mikro-orm/reproduction`:

https://github.com/sebastiaanviaene/mikro-orm-reproduction

### What driver are you using?

@mikro-orm/sqlite (reproduction)
@mikro-orm/postgresql (my project)

### MikroORM version

tried 6.1.7 & 6.2.1 in my project, the reproduction one uses next

### Node.js version

20.9.0

### Operating system

MacOS

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Sounds like something you should report to vscode instead of here. Will take a look later today, but since it works fine in webstorm and not in vscode…

**Reply by B4nan** [reply]
I recall someone on discord also said vscode was super slow for them and using next version of TS helped them.

**Reply by sebastiaanviaene** [reply]
Agree with your statement but I'm guessing you would want your library to properly work on VSCode if it's easily solved. Might also be a symptom of something else.

**Reply by sebastiaanviaene** [reply]
I just tried next version of TS [5.5.0-dev.20240430](https://www.npmjs.com/package/typescript/v/5.5.0-dev.20240430)
But nothing changes (reproduction repo)

**Reply by B4nan** [reply]
Well, yes, but I can't deal with bugs in IDEs, I have my own bugs to fix... I literally have no idea how should I debug this, speaking as someone who does not use vscode at all.

> I just tried next version of TS [5.5.0-dev.20240430](https://www.npmjs.com/package/typescript/v/5.5.0-dev.20240430)

How exactly? I am not sure if this is about installing it as part of your project deps, this will be more likely about the language server, so about the TS version your vscode is using. 

Found the comment I was referring to, its actually on reddit instead, and he refers to some vscode extension:

https://www.reddit.com/r/typescript/comments/1c67jur/comment/kzzesfk/

**Reply by sebastiaanviaene** [reply]
> How exactly? I am not sure if this is about installing it as part of your project deps, this will be more likely about the language server, so about the TS version your vscode is using.

I've set up VSCode so it matches the typescript version my project uses 

![image](https://github.com/mikro-orm/mikro-orm/assets/6652818/a8ba53f7-5633-4f7a-a02e-c8a68bc5bba0)

>Found the comment I was referring to, its actually on reddit instead, and he refers to some vscode extension:

I will try the extension and come back to you :)

**Reply by sebastiaanviaene** [reply]
No success, it's allmost the same version as well

![image](https://github.com/mikro-orm/mikro-orm/assets/6652818/60588e61-3158-4699-982b-5e0762b5e675)

**Reply by connor4312** [reply]
Hi, please see my comments here https://github.com/microsoft/vscode/issues/211805#issuecomment-2090799203

**Reply by B4nan** [reply]
Thank you! I still don't understand how come this works fine with `console.log(user)` but not with the explicit invocation of the inspect via the symbol property, but a repro is all I need, understanding can come later :D Looking into this right now.

**Reply by connor4312** [reply]
`console.log` doesn't look at `nodejs.util.inspect.custom`, that's only hit if you use `require('util').inspect(<obj>)`

**Reply by B4nan** [reply]
Uh, what? Now I am really confused, the custom inspect is surely triggered when I `console.log(user)`, just tried again.

**Reply by B4nan** [reply]
Btw this is the place where its defined:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/entity/EntityHelper.ts#L144

**Reply by B4nan** [reply]
I do call `inspect` internally, that's true, but from inside the inspect method - it needs to be triggered from outside first. It's clearly used when processing the entity itself, not just for its properties, just verified that as well locally.

**Reply by connor4312** [reply]
Oh, you're right. I didn't know that started happening.

Looks like `console.log` passes a `depth` of `2` when calling the function. The VS Code doesn't pass any depth when it's called as I implemented that before depth was added as a standard argument in Node 16. We should probably pass that too, added in https://github.com/microsoft/vscode-js-debug/pull/2007

**Reply by B4nan** [reply]
Ok, so it was all about a missing `depth`, now I get it. Did the same in the ORM, that sounds reasonable: [f706c06](https://github.com/mikro-orm/mikro-orm/commit/f706c06ff0219093644d90934d0b342ec4cbd075).

**Reply by connor4312** [reply]
yea, I'm guessing there's some cycles in the object and an undefined depth made it try to recurse infinitely

**Reply by sebastiaanviaene** [reply]
Thanks for taking care of this guys, appreciated!
